### PR TITLE
perf: replace querystring with fast-querystring

### PR DESCRIPTION
**Pull request by anonrig**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by anonrig** [comment]
@mcollina

**Comment by Eomm** [comment]
following the other prs https://github.com/fastify/fastify-reply-from/pull/272

Could you please add an option to configure the querystring module to use? Like in Fastify.

**Comment by anonrig** [comment]
@Eomm There's already an option to support this: https://github.com/fastify/fast-proxy#querystring

**Comment by anonrig** [comment]
I released 1.0 and updated the pull request.

**Review by Fdawgs — APPROVED** [review]

**Review by Uzlopak — APPROVED** [review]

**Comment by Uzlopak** [comment]
Actually the remark by eomm is not solved. queryString option is an option to replace the querystring of a request it seems. But it is not an option to replace the querystring "engine" itself.

**Comment by anonrig** [comment]
Yes, I totally missed that. I'll update the pull request. Thanks for the correction @Uzlopak

**Comment by anonrig** [comment]
I added `queryString` parameter and updated both tests and docs. Can someone approve the workflow to run?

File: index.js [hunk 1]
```diff
@@ -29,6 +29,7 @@ function fastProxy (opts = {}) {
 
   const cache = getCacheStorage(opts.cacheURLs)
   const base = opts.base
+  const queryStringModule = opts.queryString || querystring;
```
Comment (Eomm) [inline index.js:32]:
```suggestion
  const queryStringModule = opts.queryString || querystring
```

File: test/4.opts.test.js [hunk 2]
```diff
@@ -19,7 +20,11 @@ nock('http://dev.com')
 describe('fast-proxy smoke', () => {
   it('init', async () => {
     const fastProxy = require('../index')({
-      base: 'http://127.0.0.1:3000'
+      base: 'http://127.0.0.1:3000',
+      queryString: {
+        stringify: (value) => querystring.stringify(value),
+        parse: (value) => querystring.parse(value)
+      }
```
Comment (Eomm) [inline test/4.opts.test.js:27]:
```suggestion
      queryString
```

Comment (anonrig) [reply test/4.opts.test.js]: I just pushed the changes. Thank you for the review!

File: index.d.ts [hunk 3]
```diff
@@ -15,6 +20,7 @@ interface Options {
   keepAliveMsecs?: number;
   maxSockets?: number;
   rejectUnauthorized?: boolean;
+  queryString?: QueryStringModule
```
Comment (RafaelGSS) [inline index.d.ts:23]:
```suggestion
  queryString?: QueryStringModule;
```

File: index.js [hunk 4]
```diff
@@ -29,6 +29,7 @@ function fastProxy (opts = {}) {
 
   const cache = getCacheStorage(opts.cacheURLs)
   const base = opts.base
+  const queryStringModule = opts.queryString || querystring
```
Comment (RafaelGSS) [inline index.js:32]:
```suggestion
  const queryStringModule = opts.queryString || require('fast-querystring')
```

require only when needed.

File: README.md [hunk 5]
```diff
@@ -58,6 +58,20 @@ Please see: https://github.com/jkyberneees/nodejs-proxy-benchmarks
 Set the base URL for all the forwarded requests. Will be required if `http2` is set to `true`
 Note that _path will be discarded_.
 
+#### queryString
+Set the query string parser and stringifier. By default `fast-proxy` uses 
+[fast-querystring](https://npmjs.com/package/fast-querystring) module. Configuration can be changed like so:
+
+```js
+...
```
Comment (RafaelGSS) [inline README.md:66]: I believe it's better to provide the full object. Just for sake of clearness.

File: README.md [hunk 6]
```diff
@@ -58,6 +58,20 @@ Please see: https://github.com/jkyberneees/nodejs-proxy-benchmarks
 Set the base URL for all the forwarded requests. Will be required if `http2` is set to `true`
 Note that _path will be discarded_.
 
+#### queryString
+Set the query string parser and stringifier. By default `fast-proxy` uses
```
Comment (Fdawgs) [inline README.md:62]:
```suggestion
Set the query string parser and stringifier. By default `fast-proxy` uses the
```

Comment (anonrig) [reply README.md]: Thanks. Updated the pull request

File: README.md (hunk 5 above)
Comment (anonrig) [reply README.md]: Thanks @RafaelGSS I just updated the pull request

File: index.js (hunk 4 above)
Comment (anonrig) [reply index.js]: Nice optimization. Thanks!

**Comment by anonrig** [comment]
I think it's time to revive this pull request. Something is wrong with Github and requesting reviews. @RafaelGSS @Eomm Can you review it again please?

**Comment by anonrig** [comment]
I'd appreciate it if someone can approve the workflow, once again. Thank you.

**Review by Eomm — APPROVED** [review]

**Comment by Eomm** [comment]
@mcollina I cannot release this package

image

we should bump a major since there are 3 major versions that has been merged by automerge

**Pull request merged**